### PR TITLE
[Az.RecoveryServices.Backup] Adding Cross Region Restore feature

### PR DESCRIPTION
**Pull request by hiaga**
## Description
Adding Cross Region Restore feature to PowerShell 

## Checklist

- [Yes] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [Yes] The title of the PR is clear and informative
- [Yes] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [Yes] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [N/A] If applicable, the changes made in the PR have proper test coverage
- [Yes] For public API changes to cmdlets:
    - [Yes] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/727
    - [Yes] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureWorkloadPsBackupProvider.cs [hunk 1]
```diff
@@ -481,14 +486,39 @@ public RestAzureNS.AzureOperationResponse TriggerRestore()
...
+                // AzureWorkload  CRR Request
+                Logger.Instance.WriteDebug("Triggering Restore to secondary region: " + secondaryRegion);
+
+                CrossRegionRestoreRequest crrRestoreRequest = new CrossRegionRestoreRequest();
+                crrRestoreRequest.CrossRegionRestoreAccessDetails = accessToken;
+                crrRestoreRequest.RestoreRequest = triggerRestoreRequest.Properties;
+
+                var response = ServiceClientAdapter.RestoreDiskSecondryRegion(
+                    rp,
+                    "LocationNotRequired",
```
Comment (sambitratha) [inline src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureWorkloadPsBackupProvider.cs:505]: Is this value expected in service? Or can we leave it empty in this case if not required

**Review by sambitratha — DISMISSED** [review]

File: src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/IaasVm/ItemTests.cs [hunk 2]
```diff
@@ -148,5 +148,14 @@ public void TestAzureRSVaultMSI()
             TestController.NewInstance.RunPsTest(
                 _logger, PsBackupProviderTypes.IaasVm, "Test-AzureRSVaultMSI");
         }
+
+        [Fact(Skip = "to be fixed in upcoming release")]
```
Comment (isra-fel) [inline src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/IaasVm/ItemTests.cs:152]: Test case should not be skipped. Is there any way you can unblock it?

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/IaasVm/ItemTests.cs (hunk 2 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/IaasVm/ItemTests.cs:152]: Currently we're facing a service side bug. Once fixed I'll add this test. Currently I see no way to unblock this test.

File: src/RecoveryServices/RecoveryServices/ChangeLog.md [hunk 3]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Adding Cross Region Restore feature in PowerShell.
```
[the remark this answers is not in the document]
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices/ChangeLog.md]: Done.

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**